shared/health/src/main/java/io/pravega/shared/health/impl/HealthConfigImpl.java

```diff
@@ -0,0 +1,188 @@
+                HealthComponent component = new HealthComponent(name, aggregator, null);
+                roots.add(component);
+                components.put(name, component);
+                relations.put(name, new HashSet<>());
+            }
+            return this;
+        }
+
+        /**
+         * Defines that there should be a parent-child relation between a {@link HealthComponent} called 'child'
+         * and a {@link HealthComponent} called 'parent'.
+         *
+         * @param child The name of the child {@link HealthComponent}.
+         * @param parent The name of the parent {@link HealthComponent}.
+         * @return A reference to the *same* object, with updates to its internal data-structures made.
+         */
+        public Builder relation(String child, String parent) {
+            if (!components.containsKey(child) || !components.containsKey(parent)) {
+                log.warn("At least one of the components in this relation has not been defined -- aborting.");
+                return this;
+            }
+            roots.remove(components.get(child));
+            relations.get(parent).add(child);
+            return this;
+        }
+
+        /**
+         * Performs an exhaustive graph traversal to ensure (DFS) that there are no cyclic relations.
+         *
+         * @return Whether or not the specification is cycle-free.
+         * @throws Exception
+         */
+        private boolean isValid() {
+            boolean cycle = false;
+            HashMap<String, Boolean> visited = new HashMap<>();
+            for (val component : relations.entrySet()) {
+                cycle |= validate(component.getKey(), visited);
+                if (cycle) {
+                    break;
+                }
+                visited.clear();
+            }
+            return !cycle;
+        }
+
+        /**
+         * The main recursive method used for the traversal.
+         *
+         * @param name The name of the {@link HealthComponent} currently being checked.
+         * @param visited The list of {@link HealthComponent} visited so far.
+         *
+         * @return Whether or not this component has been searched previously and is cycle-free.
+         */
+        private boolean validate(String name, Map<String, Boolean> visited) {
+            if (visited.containsKey(name) && visited.get(name)) {
+                return true;
+            }
+            boolean cycle = false;
+            visited.put(name, true);
+            for (String child : relations.get(name)) {
+                cycle |= validate(child, visited);
+                if (cycle) {
+                    break;
+                }
+            }
+            return cycle;
+        }
+
+        public HealthConfig empty() {
+            HealthConfig config = null;
+            try {
+                config = build();
+            } catch (Exception e) {
+                log.error("Error building empty HealthConfig.", e);
```

Comment:
so ... will you just return `null` in this case? I don't understand how this is useful.